app/lib/bk/compat/parsers/bitrise/workflows.rb

```diff
@@ -14,10 +14,25 @@ def parse_workflow(wf_name, wf_config)
           step_key, step_config = step.first
           cmd_step << translate_step(step_key, step_config)
         end
-
+        process_workflow_envs(wf_config, cmd_step)
         cmd_step
       end
 
+      def process_workflow_envs(wf_config, cmd_step)
+        cmd_step.env ||= {}
+        (wf_config['envs'] || []).each do |env|
+          process_env_variable(env, cmd_step.env)
+        end
+      end
+
+      def process_env_variable(env, target_env)
+        env.each do |key, value|
+          next if key == 'opts' || key.to_s.strip.empty? || value.to_s.strip.empty?
+
+          target_env[key] = value
+        end
+      end
```

Comment:
1- this is the same method as in the `bitrise.rb`, it should be moved to a different file or a helper to avoid duplication
2- I have provided an alternative version that would generalize it enough to be used in both places